src/plugins/upf/upf_pfcp_api.c

```diff
@@ -905,9 +694,14 @@ handle_create_pdr (upf_session_t * sx, pfcp_create_pdr_t * create_pdr,
 	ASSERT (!pool_is_free_index (gtm->upf_apps, p[0]));
 	app = pool_elt_at_index (gtm->upf_apps, p[0]);
 	create->pdi.adr.application_id = p[0];
-	create->pdi.adr.db_id = upf_adf_get_adr_db (p[0]);
+	create->pdi.adr.db_id = upf_adf_get_adr_db (p[0], &acl);
 	create->pdi.adr.flags = app->flags;
 
+        vec_foreach (acl_rule, acl)
+          {
+            vec_add1(create->pdi.acl, *acl_rule);
+          }
+
```

Comment:
I don't think that this is valid. A PDI can contain SDFs and an Application Id. The ACL need to be merged.

Reply:
This adds some rules, not replaces them

Reply:
Fixed